src/codewhisperer/util/runtimeLanguageContext.ts

```diff
@@ -47,13 +58,51 @@ export class RuntimeLanguageContext {
         }
     }
 
-    public convertLanguage(languageId?: string) {
-        languageId = languageId === CodeWhispererConstants.typescript ? CodeWhispererConstants.javascript : languageId
-        if (!languageId) {
-            return 'plaintext'
+    /**
+     *
+     * @param vscLanguageId : official vscode languageId
+     * @returns corresponding cwspr languageId if any, otherwise fallback to vscLanguageId
+     */
+    public mapVscLanguageToCodeWhispererLanguage(vscLanguageId?: string): string {
+        if (!vscLanguageId) return CodeWhispererConstants.plaintext
+        return this.supportedLanguageMap.get(vscLanguageId) ?? vscLanguageId
+    }
+
+    /**
+     * Mapping the field ProgrammingLanguage of codewhisperer generateRecommendationRequest | listRecommendationRequest to
+     * its cwspr runtime language e.g. jsx -> typescript, typescript -> typescript
+     * @param request : cwspr generateRecommendationRequest | ListRecommendationRequest
+     * @returns request with source language name mapped to cwspr runtime language
+     */
+    public mapToRuntimeLanguage<
+        T extends codewhispererClient.ListRecommendationsRequest | codewhispererClient.GenerateRecommendationsRequest
+    >(request: T): T {
+        const fileContext = request.fileContext
+        const childLanguage = request.fileContext.programmingLanguage
+        let parentLanguage: codewhispererClient.ProgrammingLanguage
+        switch (childLanguage.languageName) {
```

Comment:
This is confusing to me because we're trying to do 2 mappings in a single step. I think it'd be better to first convert `languageName` into `CodewhispererLanguage | undefined` via `supportedLanguageMap.get` and then map that into the appropriate parent language. The less CodeWhisperer has to deal with VS Code language ids, the better.

We can also get rid of `vscodeLanguageId` as maintaining an inverse mapping is no longer needed. It's perfectly acceptable to use the literal 'jsx' string here rather than a variable.

Reply:
yea make sense